### PR TITLE
Update JdbiFactory to use metrics' InstrumentedSqlLogger

### PR DESCRIPTION
**Pull request by FredDeschenes**
###### Problem:

Metrics library was updated to use JDBI 3's `SqlLogger` instead of the deprecated `TimingCollector`, but Dropwizard is still using `TimingCollector`.

###### Solution:

Updated Dropwizard to use the new `SqlLogger`.

### CODE REVIEW

File: dropwizard-jdbi3/src/main/java/io/dropwizard/jdbi3/JdbiFactory.java [hunk 1]
```diff
@@ -15,7 +16,7 @@
 import org.jdbi.v3.sqlobject.SqlObjectPlugin;
 
 public class JdbiFactory {
-    private final StatementNameStrategy nameStrategy;
+    protected final StatementNameStrategy nameStrategy;
```
Comment (jplock) [inline dropwizard-jdbi3/src/main/java/io/dropwizard/jdbi3/JdbiFactory.java]: can this remain private?

Comment (FredDeschenes) [reply dropwizard-jdbi3/src/main/java/io/dropwizard/jdbi3/JdbiFactory.java]: Yeah I could keep it private and send it as an argument to `buildSQLLogger`.

**Comment by joschi** [comment]
@FredDeschenes Thanks for your PR!

Please make sure to run the tests after your changes.

**Comment by FredDeschenes** [comment]
Should be good now.

**Pull request merged**